Sources/xcproj/PBXProjEncoder.swift

```diff
@@ -10,14 +10,76 @@ extension PlistSerializable {
     var multiline: Bool { return true }
 }
 
+protocol ProjEncoderBuffer {
+    mutating func append(_ value: String)
+    func write(to fileURL: URL) throws
+}
+
+extension String: ProjEncoderBuffer {
+    mutating func append(_ value: String) {
+        self = self + value
+    }
+
+    func write(to fileURL: URL) throws {
+        try self.write(to: fileURL, atomically: true, encoding: .utf8)
+    }
+}
+
+extension OutputStream: ProjEncoderBuffer {
+    func append(_ value: String) {
+        switch self.streamStatus {
+        case .notOpen:
+            self.open()
+        case .closed:
+            fatalError("Cannot append to closed stream")
+            break
+        default:
+            break
+        }
+
+        let data = value.data(using: .utf8)!
+        _ = data.withUnsafeBytes{
+            self.write($0, maxLength: data.count)
+        }
+    }
+
+    func write(to fileURL: URL) throws {
+        // No-Op since we'll used
+        switch self.streamStatus {
```

Comment:
It's confusing having a method in the protocol that takes `URL` and that is not used since the `OutputStream` needs to be initialized with an URL. What about having two protocols that *"inherit"* from a base one that contains the `append` method?

Reply:
Sorry I meant to remove this, the output stream implementation wasnt better than a simple in-memory data buffer so this complexity is unnecessary.